src/validation.cpp

```diff
@@ -4984,3 +5029,33 @@ void ChainstateManager::MaybeRebalanceCaches()
         }
     }
 }
+
+CBlockIndex* ChainstateManager::getSnapshotBaseBlock()
+{
```

Comment:
> use PascalCase for the function name in new code

@ryanofsky has pioneered using `thisCovention` for method names (in order to distinguish methods from functions without using `this`) and so I've been using that here.

Reply:
There isn't a function with the name `GetSnapshotHeight` or `GetSnapshotBaseBlock` -- why would you need to distinguish the method from a function that doesn't exist? I really dislike the "let's randomly introduce new and different code style conventions" approach, especially when combined with "let's patch the entire codebase to update to the latest style conventions".

The developers-notes say "Class names, function names, and method names are UpperCamelCase
(PascalCase). Do not prefix class names with `C`."

Reply:
> There isn't a function with the name `GetSnapshotHeight` or `GetSnapshotBaseBlock` -- why would you need to distinguish the method from a function that doesn't exist?

The reason I like calling non-static member variables `m_var` instead of `var` is not to distinguish them from other variables called `var`, but to make it obvious that they are members, and make code easier to read and understand. Similarly the reason I like calling non-static member functions `fun()` instead of `Fun()` is not to distinguish from other functions call `Fun()`, but to make it obvious that they are members, so code is clearer and easier to understand. It is nice to know, when you are calling a function, if that function will have access to `*this`.

I just wanted to respond to your question though, not defend or debate anything. This could easily be a garbage coding convention and I am stupid for liking it.

Reply:
It's a completely fine convention, it's just not the one this codebase has adopted. And if we keep changing conventions we'll never get the benefit of any of them.

Reply:
> it's just not the one this codebase has adopted

@ajtowns I encourage you to take a look at the entire `src/interfaces` and `src/node` subtrees. Pretty much all of the multiprocess code (and increasingly some validation code) has been using this convention for years.

Reply:
@jonatack agree, but previous attempts have been rejected and locked (?): https://github.com/bitcoin/bitcoin/pull/14635#issuecomment-492796314

Reply:
Relented on all counts here.